src/components/core/dropdown/types/index.ts

```diff
@@ -0,0 +1,132 @@
+import { ReactNode } from 'react';
+import { ICompoundListItem } from '@core/list/types';
+
+type THintTextProps = { error: string; hint: string };
+
+type TListSize = 'small' | 'medium' | 'large';
+
+export type TDirection = 'up' | 'down';
+
+export type TDropdownBase = {
```

Comment:
Why declare a custom type like this? Instead you should make this to an interface and extend `HTMLDetailedAttributes<HTMLSelectElement`. I strongly suggest interfaces. Maybe the use rwants to extend it.

eg:
```
interface ProgressBarProps extends HTMLAttributes<HTMLSelectElement> {
```